models/public/gmcnn-tf/freeze_model.py

```diff
@@ -0,0 +1,53 @@
+import argparse
+from pathlib import Path
+import tensorflow as tf
+
+from net.network import GMCNNModel
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Freeze GMCNN Model')
+    parser.add_argument("--ckpt_dir", type=str, required=True,
+                        help="Path to pretrained checkpoints.")
+    parser.add_argument("--save_dir", type=Path, required=True,
+                        help="Path to write frozen graph.")
+    return parser.parse_args()
+
+class Options:
+    img_shapes = [512, 680]
+    mask_type = 'free_form'
+    g_cnum = 32
+    d_cnum = 64
+
+def freeze_model(config, input_checkpoints, output_dir):
+    model = GMCNNModel()
+
+    with tf.Session() as sess:
+        input_image = tf.placeholder(dtype=tf.float32, shape=[None, *config.img_shapes, 3])
+        input_mask = tf.placeholder(dtype=tf.float32, shape=[None, *config.img_shapes, 1])
+
+        output = model.evaluate(input_image, input_mask, config)
+        output = (output + 1) * 127.5
+        output = tf.minimum(tf.maximum(output[:, :, :, ::-1], 0), 255)
+
+        vars_list = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES)
+        assign_ops = list(map(lambda x: tf.assign(x, tf.train.load_variable(input_checkpoints, x.name)),
```

Comment:
`map` with a `lambda` is a sign that you should use a list comprehension instead. :-)